components/ai_chat/resources/page/state/data-context-provider.tsx

```diff
@@ -292,7 +292,7 @@ function DataContextProvider (props: DataContextProviderProps) {
     suggestedQuestions,
     canGenerateQuestions,
     userAutoGeneratePref,
-    siteInfo: siteInfo || null,
```

Comment:
this was the problem. `<Main` was waiting for `siteInfo` to not be `undefined`, but we were hiding the `undefined` and forcing it to `null`.